molecularnodes/io/parse/star.py

```diff
@@ -19,90 +56,166 @@ def _n_images(self):
+            if not pot_micrograph_path.exists():
+                if self.star_type == 'relion':
+                    pot_micrograph_path = Path(self.file_path).parent.parent.parent / micrograph_path
+                    if not pot_micrograph_path.exists():
+                        raise FileNotFoundError(f"Micrograph file {micrograph_path} not found")
+                else:
+                    raise FileNotFoundError(f"Micrograph file {micrograph_path} not found")
+            micrograph_path = pot_micrograph_path
+
+        tiff_path = Path(micrograph_path).with_suffix('.tiff')
+        if not tiff_path.exists():
+            with mrcfile.open(micrograph_path) as mrc:
+                micrograph_data = mrc.data.copy()
+
+            # For 3D data sum over the z axis. Probalby would be nicer to load the data as a volume
+            if micrograph_data.ndim == 3:
+                micrograph_data = np.sum(micrograph_data, axis=0)
+            # Normalize the data to 0-1
+            micrograph_data = (micrograph_data - micrograph_data.min()) / (micrograph_data.max() - micrograph_data.min())
+            
+            if micrograph_data.dtype != np.float32:
+                micrograph_data = micrograph_data.astype(np.float32)
+            from PIL import Image
+            # Need to invert in Y to generate the correct tiff
+            Image.fromarray(micrograph_data[::-1,:]).save(tiff_path)
+        return tiff_path
+    
+    def _update_micrograph_texture(self, *_):
+        import bpy
+        try:
+            show_micrograph = self.star_node.inputs['Show Micrograph']
+            _ = self.object['mn']
+        except ReferenceError:
+            bpy.app.handlers.depsgraph_update_post.remove(self._update_micrograph_texture)
+            return
+        if self.star_node.inputs['Image'].default_value == self.current_image:
+            return
+        else:
+            self.current_image = self.star_node.inputs['Image'].default_value
+        if not show_micrograph:
+            return
+        tiff_path = self._convert_mrc_to_tiff()
+        if tiff_path:
+            try:
+                image_obj = bpy.data.images[tiff_path.name]
+            except KeyError:
+                image_obj = bpy.data.images.load(str(tiff_path))
+            image_obj.colorspace_settings.name = 'Non-Color'
+            self.micrograph_material.node_tree.nodes['Image Texture'].image = image_obj
+            self.star_node.inputs['Micrograph'].default_value = image_obj
+
+                 
 
-        object = bl.obj.create_object(
-            xyz * world_scale, collection=bl.coll.mn(), name=name)
-
-        object.mn['molecule_type'] = 'star'
-        object.mn['star_type'] = star_type
-
-        # create the attribute and add the data for the image id
-        bl.obj.set_attribute(object, 'MNImageId', image_id, 'INT', 'POINT')
+    def create_model(self, name='StarFileObject', node_setup=True, world_scale=0.01):
+        import bpy
```

Comment:
these imports can go to the top of the file. The `bpy` is already imported there